packages/test/test-end-to-end-tests/src/test/gc/gcTestSummaryUtils.ts

```diff
@@ -0,0 +1,167 @@
+    return fluidObject.ISummarizer;
+}
+
+const defaultSummaryOptions: ISummaryRuntimeOptions = {
+    summaryConfigOverrides: {
+        state: "disableHeuristics",
+        maxAckWaitTime: 10000,
+        maxOpsSinceLastSummary: 7000,
+        initialSummarizerDelayMs: 0,
+        summarizerClientElection: false,
+    },
+};
+
+export async function createSummarizerFromFactory(
+    provider: ITestObjectProvider,
+    container: IContainer,
+    dataStoreFactory: IFluidDataStoreFactory,
+    summaryVersion?: string,
+    containerRuntimeFactoryType = ContainerRuntimeFactoryWithDefaultDataStore,
+): Promise<ISummarizer> {
+    const innerRequestHandler = async (request: IRequest, runtime: IContainerRuntimeBase) =>
+        runtime.IFluidHandleContext.resolveHandle(request);
+    const runtimeFactory = new containerRuntimeFactoryType(
+        dataStoreFactory,
+        [
+            [dataStoreFactory.type, Promise.resolve(dataStoreFactory)],
+        ],
+        undefined,
+        [innerRequestHandler],
+        { summaryOptions: defaultSummaryOptions },
+    );
+
+    const loader = provider.createLoader(
+        [[provider.defaultCodeDetails, runtimeFactory]],
+        { configProvider: mockConfigProvider() },
+    );
+
+    return createSummarizerCore(container, loader, summaryVersion);
+}
+
+export async function createSummarizer(
+    provider: ITestObjectProvider,
+    container: IContainer,
+    summaryVersion?: string,
+    gcOptions?: IGCRuntimeOptions,
+): Promise<ISummarizer> {
+    const testContainerConfig: ITestContainerConfig = {
+        runtimeOptions: {
+            summaryOptions: defaultSummaryOptions,
+            gcOptions,
+        },
+        loaderProps: { configProvider: mockConfigProvider() },
+    };
+
+    const loader = provider.makeTestLoader(testContainerConfig);
+    return createSummarizerCore(container, loader, summaryVersion);
+}
+
+export async function summarizeNow(summarizer: ISummarizer, reason: string = "end-to-end test") {
+    const result = summarizer.summarizeOnDemand({ reason });
+
+    const submitResult = await result.summarySubmitted;
+    assert(submitResult.success, "on-demand summary should submit");
+    assert(submitResult.data.stage === "submit",
+        "on-demand summary submitted data stage should be submit");
+    assert(submitResult.data.summaryTree !== undefined, "summary tree should exist");
+
+    const broadcastResult = await result.summaryOpBroadcasted;
+    assert(broadcastResult.success, "summary op should be broadcast");
+
+    const ackNackResult = await result.receivedSummaryAckOrNack;
+    assert(ackNackResult.success, "summary op should be acked");
+
+    await new Promise((resolve) => process.nextTick(resolve));
+
+    return {
+        summaryTree: submitResult.data.summaryTree,
+        summaryVersion: ackNackResult.data.summaryAckOp.contents.handle,
+    };
+}
+
+export function getGCStateFromSummary(summary: ISummaryTree): IGarbageCollectionState | undefined {
```

Comment:
Hmmm, we use this in the actual GC function, I wonder if this should be exported from a more public package